### PR TITLE
alibi-detect 0.11.4 :snowflake:

### PR DESCRIPTION
**Pull request by lorepirri**
alibi-detect 0.11.4 :snowflake:

**Destination channel:** Snowflake

### Links

- [PKG-5102]
- dev_url (pypi): https://github.com/SeldonIO/alibi-detect/tree/v0.11.4
- conda-forge: https://github.com/conda-forge/alibi-detect-feedstock/blob/main/recipe/meta.yaml
- pypi: https://pypi.org/project/alibi-detect/0.11.4
- pypi inspector: https://inspector.pypi.io/project/alibi-detect/0.11.4

### Explanation of changes:

- bump version, mirror conda-forge
- patch because the function `scipy.stats.binom_test()` was removed from SciPy 1.12.0, see https://stackoverflow.com/questions/78090426/cannot-import-name-binom-test-from-scipy-stats-error-when-using-borutashap
- ~patch because `opt_dep` attribute is not there when the tests are not run in the CI
and a runtime error occurs.~
- use  `opt_dep` in tests to test the optional dependencies.

[PKG-5102]: https://anaconda.atlassian.net/browse/PKG-5102?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -6,60 +6,90 @@ package:
...
     # Core dependencies
-    - python >=3.7
+    - python
     - dill >=0.3.0,<0.4.0
     - matplotlib-base >=3.0.0,<4.0.0
+    # avoid 0.54 due to: https://github.com/SeldonIO/alibi/issues/466
     - numba >=0.50.0,!=0.54.0,<0.56.0
     - numpy >=1.16.2,<2.0.0
-    - opencv >=3.2.0,<5.0.0
+    - py-opencv >=3.2.0,<5.0.0
```
Comment (lorepirri) [inline recipe/meta.yaml]:
```suggestion
    - opencv >=3.2.0,<5.0.0
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -6,60 +6,90 @@ package:
...
+    - setuptools
+    - wheel
   run:
     # Core dependencies
-    - python >=3.7
+    - python
     - dill >=0.3.0,<0.4.0
     - matplotlib-base >=3.0.0,<4.0.0
+    # avoid 0.54 due to: https://github.com/SeldonIO/alibi/issues/466
     - numba >=0.50.0,!=0.54.0,<0.56.0
```
Comment (lorepirri) [inline recipe/meta.yaml]:
```suggestion
    - numba >=0.50.0,!=0.54.0
```

**Review by ViridianMelody — APPROVED** [review]
Looks great!

File: recipe/patches/0001-fix-binom_test-to-binomtest.patch [hunk 3]
```diff
@@ -0,0 +1,1040 @@
...
++from scipy.stats import binomtest, ks_2samp
+ from sklearn.model_selection import StratifiedKFold
+ 
+ if has_pytorch:
+@@ -20,24 +20,24 @@ logger = logging.getLogger(__name__)
+ 
+ 
+ class BaseClassifierDrift(BaseDetector):
+-    model: Union['tf.keras.Model', 'torch.nn.Module']
++    model: Union["tf.keras.Model", "torch.nn.Module"]
```
Comment (JeanChristopheMorinPerso) [inline recipe/patches/0001-fix-binom_test-to-binomtest.patch]: A lot of these changes are just formatting changes. Can you undo these and only change what's needed please?

File: recipe/patches/0002-fix-opt_dep-attribute-check.patch [hunk 4]
```diff
@@ -0,0 +1,28 @@
+From d01d2dea1875a302e73a37a23bccf53114fd4133 Mon Sep 17 00:00:00 2001
+From: Lorenzo Pirritano <lpirritano@anaconda.com>
+Date: Tue, 25 Jun 2024 14:36:30 +0200
+Subject: [PATCH 2/2] fix opt_dep attribute check
+
+opt_dep is not there when the tests are not run in the CI
```
Comment (JeanChristopheMorinPerso) [inline recipe/patches/0002-fix-opt_dep-attribute-check.patch]: You can pass `--opt-dep` to pytest. The function says:
>    Tests that use this fixture must be run with the --opt-dep option via terminal. If not they will skip. This fixture is used in CI to indicate the optional dependencies installed in the tox environments the tests are run in. See setup.cfg and .github/workflows/ci.yml for more details.

Also see https://github.com/SeldonIO/alibi-detect/blob/master/setup.cfg#L66-L127.

**Review by JeanChristopheMorinPerso — CHANGES_REQUESTED** [review]

File: recipe/patches/0001-fix-binom_test-to-binomtest.patch (hunk 3 above)
Comment (lorepirri) [reply recipe/patches/0001-fix-binom_test-to-binomtest.patch]: Ah sorry, I forgot to disable the formatting. Done, thanks for the comment!

File: recipe/patches/0002-fix-opt_dep-attribute-check.patch (hunk 4 above)
Comment (lorepirri) [reply recipe/patches/0002-fix-opt_dep-attribute-check.patch]: Somehow using `--pyargs` made it not possible to pass `--opt-dep`, so I removed `--pyargs` and used `--opt-dep=all`. So, now we can run tests for all optional dependencies (`tensorflow`, `prophet`, `pytorch`, except `pykeops` which we do not have). Thank you, your comment helped improving the tests.

**Comment by lorepirri** [comment]
will need to re-fix the pinning tomorrow.

**Comment by skupr-anaconda** [comment]
@lorepirri An error on `linux-64`, `linux-aarch64`,`osx-arm64` and `win-64`:
`FAILED alibi_detect/tests/test_dep_management.py::test_od_dependencies` and `AssertionError: OutlierProphet has been imported as an instance of MissingDependency. Are you sure the dependency buckets, ['prophet'] are correct?`


Also, there are incompatible dependencies on `osx-64`. It can be `blas * openblas` or `opencv`.

**Comment by lorepirri** [comment]
thank you @skupr-anaconda 

> An error on `linux-64`, `linux-aarch64`,`osx-arm64` and `win-64`:
> `FAILED alibi_detect/tests/test_dep_management.py::test_od_dependencies` and `AssertionError: OutlierProphet has been imported as an instance of MissingDependency. Are you sure the dependency buckets, ['prophet'] are correct?`

this one is because I removed prophet from the tests and the --opt-dep=all

it's ok, just a test. I will enable only tensorflow now

> Also, there are incompatible dependencies on osx-64. It can be blas * openblas or opencv.

that is the problem that I am trying to fix by trial and error.

**Review by JeanChristopheMorinPerso — APPROVED** [review]

File: recipe/meta.yaml [hunk 5]
```diff
@@ -82,10 +136,8 @@ about:
     of drift, how it can be detected in a principled manner and also describes the 
     anatomy of a drift detector.
 
-
     PyPI: [https://pypi.org/project/alibi-detect/](https://pypi.org/project/alibi-detect/)
-
-  doc_url: https://docs.seldon.io/projects/alibi-detect/en/latest/
+  doc_url: https://docs.seldon.io/projects/alibi-detect/en/latest
```
Comment (ifitchet) [inline recipe/meta.yaml]:
```suggestion
  doc_url: https://docs.seldon.io/projects/alibi-detect
```

File: recipe/meta.yaml [hunk 6]
```diff
@@ -6,60 +6,114 @@ package:
...
-  # commands:
-  #   - pip check
-  # requires:
-  #   - pip
+  requires:
+    - pip
+    - pytest >=5.3.5,<8.0.0
+    - pytest-xdist >=1.28.0,<4.0.0
+    - pytest-randomly >=3.5.0,<4.0.0
+    - pytest-timeout >=1.4.2,<3.0.0
```
Comment (ifitchet) [inline recipe/meta.yaml:74]: Of interest, where are these pinnings defined?

**Comment by ifitchet** [comment]
You mention two patches in the Explanation of changes, one of them has gone?

**Comment by lorepirri** [comment]
> You mention two patches in the Explanation of changes, one of them has gone?

updated the description, thanks.

yes one is gone in favor of using the parameter `opt_dep`

File: recipe/meta.yaml (hunk 6 above)
Comment (lorepirri) [reply recipe/meta.yaml:74]: https://github.com/SeldonIO/alibi-detect/blob/7cf1665112e2fb82d3ed47c941d7e7ed7b9f921a/requirements/dev.txt#L7-L12

Comment (ifitchet) [reply recipe/meta.yaml:74]: Oh, interesting.
My tool was looking in the pypi tarball and not finding any `requirements*` files or directories.  That might need a rethink.

**Review by ifitchet — APPROVED** [review]
LGTM

**Review by skupr-anaconda — APPROVED** [review]

File: recipe/meta.yaml (hunk 6 above)
Comment (lorepirri) [reply recipe/meta.yaml:74]: Well the tarball sometimes is a stripped-down re-org of the release archive. We probably always need to have a look at the repository.

**Pull request merged**